app/web/template/layout/date.gohtml

```diff
@@ -2,17 +2,27 @@
   <div class="govuk-form-group {{ if index .top.Errors .name }}govuk-form-group--error{{ end }} govuk-!-margin-bottom-6">
     <fieldset class="govuk-fieldset" role="group" aria-describedby="{{ .name }}-hint">
       <legend class="govuk-fieldset__legend govuk-fieldset__legend--m">{{ tr .top.App .label }}</legend>
+      {{ if .context }}
+        <p class="govuk-body">
+          {{ tr .top.App .context }}
+        </p>
+      {{ end }}
       {{ if .hint }}
         <div id="{{ .name }}-hint" class="govuk-hint">
           {{ tr .top.App .hint }}
         </div>
       {{ end }}
+
       {{ template "error-message" (errorMessage .top .name) }}
+      {{ template "error-message" (errorMessage .top (printf "%v-day" .name)) }}
+      {{ template "error-message" (errorMessage .top (printf "%v-month" .name)) }}
+      {{ template "error-message" (errorMessage .top (printf "%v-year" .name)) }}
+
       <div class="govuk-date-input">
         <div class="govuk-date-input__item">
           <div class="govuk-form-group">
-            <label class="govuk-label govuk-date-input__label" for="f-{{ .name }}">{{ tr .top.App "day" }}</label>
-            <input class="govuk-input govuk-date-input__input govuk-input--width-2 {{ if index .top.Errors .name }}govuk-input--error{{ end }}" id="f-{{ .name }}" name="{{ .name }}-day" type="text" pattern="[0-9]*" inputmode="numeric" value="{{ .value.Day }}" />
+            <label class="govuk-label govuk-date-input__label" for="f-{{ .name }}-day">{{ tr .top.App "day" }}</label>
```

Comment:
this didn't have `-day` so that clicking the error for when it is missing would take you to the field

Reply:
Makes sense, this allows a user to go to the individual field when a day, month, or year error is thrown but breaks the general Enter a date linking. Something to think about in the validation ticket.